### PR TITLE
refactor(sbb-slider): lit migration

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- sbb-slider migrated to Lit

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [x] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/components/sbb-slider/sbb-slider.tsx [hunk 1]
```diff
@@ -1,86 +1,92 @@
...
     this._handleChange();
   }
 
-  @Watch('value')
-  @Watch('valueAsNumber')
-  private _syncValues(newValue: string | number = this._rangeInput?.valueAsNumber): void {
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('value') && !!changedProperties.get('value')) {
+      this._handleChange(Number(this.value));
+    } else if (changedProperties.has('valueAsNumber') && !!changedProperties.get('valueAsNumber')) {
```
Comment (jeripeierSBB) [inline src/components/sbb-slider/sbb-slider.tsx:84]: maybe I'm misunderstanding, but what if you like to set to 0 ?

File: src/components/sbb-slider/sbb-slider.tsx [hunk 2]
```diff
@@ -1,86 +1,92 @@
...
+    const signal = this._abort.signal;
+    this.addEventListener('keydown', (e) => this._handleKeydown(e), { signal });
     this._handleChange();
   }
 
-  @Watch('value')
-  @Watch('valueAsNumber')
-  private _syncValues(newValue: string | number = this._rangeInput?.valueAsNumber): void {
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('value') && !!changedProperties.get('value')) {
```
Comment (jeripeierSBB) [inline src/components/sbb-slider/sbb-slider.tsx:82]: maybe I'm misunderstanding, but what if you like to set to 0 ?

File: src/components/sbb-slider/sbb-slider.tsx [hunk 3]
```diff
@@ -117,12 +121,13 @@ export class SbbSlider implements ComponentInterface {
...
       isNaN(mathFraction) || mathFraction < 0 ? 0 : mathFraction > 1 ? 1 : mathFraction;
-    this._syncValues();
+    this._syncValues(value);
   }
 
-  @Listen('keydown')
-  public async handleKeydown(event: KeyboardEvent): Promise<void> {
-    if (this.disabled || this.readonly || (await isEventPrevented(event))) {
+  private async _handleKeydown(event: KeyboardEvent): Promise<void> {
+    // FIXME the third condition in the if doesn't prevent from scrolling
```
Comment (jeripeierSBB) [inline src/components/sbb-slider/sbb-slider.tsx:128]: Can't it be fixed? "Third condition" would not be a good hint for what to do later

File: src/components/sbb-slider/sbb-slider.tsx [hunk 4]
```diff
@@ -1,86 +1,92 @@
...
-} from '@stencil/core';
-import { isEventPrevented, forwardEventToHost } from '../../global/eventing';
+import { forwardEventToHost, EventEmitter, ConnectedAbortController } from '../../global/eventing';
+import { CSSResult, html, LitElement, nothing, TemplateResult, PropertyValues } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { setAttributes } from '../../global/dom';
+import { ref } from 'lit/directives/ref.js';
+import Style from './sbb-slider.scss?lit&inline';
+import { spread } from '@open-wc/lit-helpers';
+import { styleMap } from 'lit/directives/style-map.js';
```
Comment (jeripeierSBB) [inline src/components/sbb-slider/sbb-slider.tsx:8]: import to sbb-icon should be added as it is used in shadow dom

File: src/components/sbb-slider/sbb-slider.tsx [hunk 5]
```diff
@@ -117,12 +121,13 @@ export class SbbSlider implements ComponentInterface {
...
-    this._syncValues();
+    this._syncValues(value);
   }
 
-  @Listen('keydown')
-  public async handleKeydown(event: KeyboardEvent): Promise<void> {
-    if (this.disabled || this.readonly || (await isEventPrevented(event))) {
+  private async _handleKeydown(event: KeyboardEvent): Promise<void> {
+    // FIXME the third condition in the if doesn't prevent from scrolling
+    event.preventDefault();
```
Comment (jeripeierSBB) [inline src/components/sbb-slider/sbb-slider.tsx:129]: Don't we change the logic here?

File: src/components/sbb-slider/sbb-slider.stories.tsx [hunk 6]
```diff
@@ -16,7 +19,7 @@ const TemplateSbbSlider = (args): JSX.Element => <sbb-slider {...args}></sbb-sli
 const TemplateSbbSliderChangeEvent = (args): JSX.Element => (
   <Fragment>
     <sbb-slider {...args} onChange={(event) => changeEventHandler(event)}></sbb-slider>
-    <div style={{ 'margin-block-start': '2rem' }}>Change slider position:</div>
+    <div style="margin-block-start: 2rem;">Change slider position:</div>
```
Comment (jeripeierSBB) [inline src/components/sbb-slider/sbb-slider.stories.tsx:22]: Why is this change needed?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Some minor fixed needed, thanks for the work!

File: src/components/sbb-slider/sbb-slider.tsx (hunk 3 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-slider/sbb-slider.tsx:128]: This was meant to be discussed but i forgot to add a comment this morning :'(

Apparently, the `(await isEventPrevented(event))` is not working anymore (maybe something related to the removal of the Listen annotation?), and fo fixing it I added a `event.preventDefault` before the if condition; if this sounds ok to you, the fixme can be simply removed.

File: src/components/sbb-slider/sbb-slider.tsx (hunk 5 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-slider/sbb-slider.tsx:129]: Marking as resolved, but discussion thread is still open in the other comment https://github.com/lyne-design-system/lyne-components/pull/2082#discussion_r1362142375

File: src/components/sbb-slider/sbb-slider.tsx (hunk 2 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-slider/sbb-slider.tsx]: Changed to changedProperties.get('<prop>') !== undefined

File: src/components/sbb-slider/sbb-slider.tsx (hunk 1 above)
Comment (jeripeierSBB) [reply src/components/sbb-slider/sbb-slider.tsx]: Should maybe null check be added as well? I don't really know the scenario

File: src/components/sbb-slider/sbb-slider.stories.tsx (hunk 6 above)
Comment (jeripeierSBB) [reply src/components/sbb-slider/sbb-slider.stories.tsx:22]: It's just before this would have been a type error or something (but at the moment deactivated), and as we did not change it else, it's maybe better to keep the old notation until we migrate stories anyways

File: src/components/sbb-slider/sbb-slider.tsx (hunk 1 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-slider/sbb-slider.tsx]: The case i would avoid is the component init, when all the props are set to undefined
image

Comment (jeripeierSBB) [reply src/components/sbb-slider/sbb-slider.tsx]: ok, thanks!

File: src/components/sbb-slider/sbb-slider.tsx (hunk 3 above)
Comment (kyubisation) [reply src/components/sbb-slider/sbb-slider.tsx:132]: From what I remember, with the native slider/range input, if the keydown event is prevented, the resulting change is prevented.

Comment (DavideMininni-Fincons) [reply src/components/sbb-slider/sbb-slider.tsx:132]: As it is now, when i change values with arrows I can see both the input/change events in the Storybook 'Actions' tab

File: src/components/sbb-slider/sbb-slider.tsx (hunk 1 above)
Comment (DavideMininni-Fincons) [reply src/components/sbb-slider/sbb-slider.tsx]: After a check with Tommaso on the sbb-checkbox, in which I placed the same logic, it seems that it's useless and possibly bugged, so i removed the undefined check.

**Review by TomMenga — APPROVED** [review]
LGTM 👍 
Wait also for @jeripeierSBB to approve since I'm not sure I understood the last open change request
